propagation/trace_context.go

```diff
@@ -139,7 +137,10 @@ func (tc TraceContext) extract(carrier TextMapCarrier) trace.SpanContext {
 	// Clear all flags other than the trace-context supported sampling bit.
 	scc.TraceFlags = trace.TraceFlags(opts[0]) & trace.FlagsSampled
 
-	scc.TraceState = parseTraceState(carrier.Get(tracestateHeader))
+	// Ignore the error returned here. Failure to parse tracestate MUST NOT
+	// affect the parsing of traceparent according to the W3C tracecontext
+	// specification.
+	scc.TraceState, _ = trace.ParseTraceState(carrier.Get(tracestateHeader))
```

Comment:
There are many situations where users of this function will want to know if the TraceState they are trying to construct is valid For example, creating a new TraceState and starting to propagate it, or [testing](https://github.com/open-telemetry/opentelemetry-go/pull/1931/files#diff-eff24bc56fe96b7105a5fdc50b564250016dcacf1f6d59524263392ea441fdd4R40). Without returning an error with this it will be impossible to tell if the function correctly parsed a zero value TraceState or errored.

Additionally, having this signature seems in line with what I would say are Go "best practices". The parsing logic is isolated from the use and it is up to the caller how they want to handle errors, instead of the function making that decision for the caller and potentially making the wrong decision.